drivers/platform/stm32/stm32_spi.c

```diff
@@ -205,30 +206,33 @@ int32_t stm32_spi_write_and_read(struct spi_desc *desc,
 				 uint8_t *data,
 				 uint16_t bytes_number)
 {
-	int32_t	ret;
+	uint8_t *tx = data;
+	uint8_t *rx = data;
 	struct stm32_spi_desc *sdesc;
-
+	struct stm32_gpio_desc *gdesc;
```

Comment:
it seems that gpio is not used anymore so it can be removed from other places too:
For example https://github.com/analogdevicesinc/no-OS/blob/62e4fa56b595be7a6c415f14a48a9202310547a2/drivers/platform/stm32/stm32_spi.h#L68
And set https://github.com/analogdevicesinc/no-OS/blob/62e4fa56b595be7a6c415f14a48a9202310547a2/drivers/platform/stm32/stm32_spi.h#L55
as uint32_t

Reply:
Actually it is still used.
I use the STM32 HAL to initialize the GPIO but then use the 'desc->extra->chip_select' to access the registers of the gpio.